protocol/tx/data.go

```diff
@@ -0,0 +1,43 @@
+package tx
+
+import (
+	"chain/crypto/sha3pool"
+	"chain/protocol/bc"
+)
+
+// A "Data" entry represents some arbitrary data
+// the transaction author wants the current transaction to commit to,
+// either for use in programs in the current or future transactions,
+// or for reference by external systems.
+// This is done with a hash commitment:
+// the entry itself stores a 32-byte hash of the underlying data,
+// which may be of any length.
+// It is the responsibility of the transport layer
+// to provide the underlying data
+// alongside the actual transaction, if necessary.
+// The data need not be made available to all parties;
+// it is fine to keep it confidential.
+//
+// Note that the body of this entry is a hash (of the underlying data);
+// the body_hash is a hash of that hash.
```

Comment:
I think it means that the hashed representation is a hash of `body` (which is a hash of the underlying data). I'll try to improve this comment.